### PR TITLE
Add Bayesian VAR(2) example script #1658

### LINKED ISSUE #1658: Could you please post a application tutorial about Baysian Vector Autoregression?
NumPyro is quite a good package about probabilistic programming, I have seen there is a tutorial about AR(2). 

Cause VAR is quite important in economics, wish you and your team could post a tutorial about this. :D

### PR DESCRIPTION
**Pull request by aibit0111**
VAR(2) examples:
Reference - https://otexts.com/fpp2/VAR.html
fix #1658

### CODE REVIEW

**Comment by tillahoffmann** [comment]
This looks interesting! As of #1904, there is a Gaussian state space model distribution. Would that be suitable for the VAR here?

**Comment by fehiepsi** [comment]
In addition, could you add the example to docs together with thumbnails (like in https://github.com/pyro-ppl/numpyro/pull/1429)?

**Comment by aibit0111** [comment]
Hi @fehiepsi, added the index and thumbnails.

**Comment by aibit0111** [comment]
> This looks interesting! As of #1904, there is a Gaussian state space model distribution. Would that be suitable for the VAR here?

Hi @tillahoffmann, awesome work on SSM, do you want me to create example using SSM?

**Comment by aibit0111** [comment]
@fehiepsi , Lint has failed, when I am fixing with ruff, it is changing some more files other that var2.py
I have been working on some econometric models with numpyro, will be adding more models.

**Comment by tillahoffmann** [comment]
> Hi @tillahoffmann, awesome work on SSM, do you want me to create example using SSM?

I think using the current implementation in this PR or SSM could both work. The latter might be a bit more concise because you wouldn't have to implement the transition function.

**Comment by fehiepsi** [comment]
@aibit0111 Could you merge with master? lint issues are fixed upstream.

**Comment by aibit0111** [comment]
@fehiepsi Done

**Comment by fehiepsi** [comment]
Could you add some docstring like other examples to give the readers a take on the content?

**Comment by aibit0111** [comment]
Hi @fehiepsi, i was on holiday, updated the code with proper doc and passed all the test. 
image

File: examples/var2.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        "Phi2", dist.Normal(0, 1).expand([K, K])
+    )  # Coefficients for lag 2
+
+    # Priors for error terms
+    with numpyro.plate("K", K):
+        sigma = numpyro.sample("sigma", dist.HalfNormal(1.0))  # Standard deviations
+    L_omega = numpyro.sample(
+        "L_omega", dist.LKJCholesky(dimension=K, concentration=1.0)
+    )
+    L_Sigma = jnp.matmul(jnp.diag(sigma), L_omega)
```
Comment (fehiepsi) [inline examples/var2.py:67]: maybe `jnp.einsum("...i,...ij->...ij", sigma, L_omega)` or `sigma[..., None] * L_omega` to improve the performance

**Comment by aibit0111** [comment]
Thanks @fehiepsi for generous review, I am new to num pyro, learning a lot. Please review once again

**Pull request merged**